### PR TITLE
[gen_l10n] Throw error when arb file does not exist

### PR DESCRIPTION
**Pull request by thkim1011**
Currently, attempting to generate localization files checks for permissions on the template arb file but does not check whether the file exists or not. This PR changes this so that a more helpful error is thrown.

#106139

## Pre-launch Checklist

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [ ] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter_tools/lib/src/localizations/gen_l10n.dart [hunk 1]
```diff
@@ -942,10 +942,9 @@ class LocalizationsGenerator {
   @visibleForTesting
   static File templateArbFileFromFileName(String templateArbFileName, Directory inputDirectory) {
     final File templateArbFile = inputDirectory.childFile(templateArbFileName);
-    final FileStat templateArbFileStat = templateArbFile.statSync();
-    final String templateArbFileStatModeString = templateArbFileStat.modeString();
-    final FileSystemEntityType templateArbFileStatType = templateArbFileStat.type;
-    if (templateArbFileStatType == FileSystemEntityType.notFound) {
+    final String templateArbFileStatModeString = templateArbFile.statSync().modeString();
+    final bool templateArbFileExists = templateArbFile.existsSync();
+    if (!templateArbFileExists) {
```
Comment (HansMuller) [inline packages/flutter_tools/lib/src/localizations/gen_l10n.dart:947]: Could just be `if (!templateArbFile.existsSync())`

File: packages/flutter_tools/lib/src/localizations/gen_l10n.dart [hunk 2]
```diff
@@ -942,10 +942,9 @@ class LocalizationsGenerator {
   @visibleForTesting
   static File templateArbFileFromFileName(String templateArbFileName, Directory inputDirectory) {
     final File templateArbFile = inputDirectory.childFile(templateArbFileName);
-    final FileStat templateArbFileStat = templateArbFile.statSync();
-    final String templateArbFileStatModeString = templateArbFileStat.modeString();
-    final FileSystemEntityType templateArbFileStatType = templateArbFileStat.type;
-    if (templateArbFileStatType == FileSystemEntityType.notFound) {
+    final String templateArbFileStatModeString = templateArbFile.statSync().modeString();
```
[the remark this answers is not in the document]
Comment (thkim1011) [reply packages/flutter_tools/lib/src/localizations/gen_l10n.dart:945]: It's used in line 952 to check the permissions of the file, I think. I can put it right before that if-block if it makes it clearer.

File: packages/flutter_tools/lib/src/localizations/gen_l10n.dart (hunk 1 above)
Comment (HansMuller) [reply packages/flutter_tools/lib/src/localizations/gen_l10n.dart]: Sorry, I was looking at the diffs not at the overall context. The version of the code you had originally, that just relied on one stat call made sense as it was. Sorry about the (my) confusion.

**Review by HansMuller — APPROVED** [review]
LGTM

**Pull request merged**